### PR TITLE
API for retrieving TSVConn from TSHttpSsn

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

**Comment by masaori335** [comment]
Looks good. But we need API Review on dev@trafficserver before merge this, because this is new TS API. Could you start API Review Process ? ( or I'm missing something? )

- https://cwiki.apache.org/confluence/display/TS/API+Review+Process

**Comment by persiaAziz** [comment]
Proposal submitted. Also added the doc

**Comment by persiaAziz** [comment]
Done

File: doc/developer-guide/api/functions/TSVConn.en.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+   permissions and limitations under the License.
+
+.. include:: ../../../common.defs
+
+.. default-domain:: c
+
+TSVConn
+*******
+
+Traffic Server APIs to get TSVConn from TSHttpSsn object
```
Comment (SolidWallOfCode) [inline doc/developer-guide/api/functions/TSVConn.en.rst]:
```
:type:`TSVConn`
```
Also for `TSHttpSsn`.

File: doc/developer-guide/api/functions/TSVConn.en.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+`#include <ts/ts.h>`
+
+.. function:: TSVConn TSHttpSsnClientVConnGet(TSHttpSsn ssnp)
+.. function:: TSVConn TSHttpSsnServerVConnGet(TSHttpSsn ssnp)
+
+Description
+===========
+
+These APIs allow the developer to get the NetVconnection (represented by TSVConn) from the Http session (TSHttpSsn) object.
```
Comment (SolidWallOfCode) [inline doc/developer-guide/api/functions/TSVConn.en.rst]: And here.

File: proxy/InkAPI.cc [hunk 3]
```diff
@@ -4559,6 +4559,20 @@ TSHttpSsnTransactionCount(TSHttpSsn ssnp)
   return cs->get_transact_count();
 }
 
+TSVConn
+TSHttpSsnClientVConnGet(TSHttpSsn ssnp)
+{
+  ProxyClientSession *cs = reinterpret_cast<ProxyClientSession *>(ssnp);
+  return cs->get_netvc() ? reinterpret_cast<TSVConn>(cs->get_netvc()) : nullptr;
```
Comment (SolidWallOfCode) [inline proxy/InkAPI.cc]: You should be able to `reinterpret_cast` `nullptr` so
```
return reinterpret_cast<TSVConn>(cs->get_netvc());
```

**Comment by persiaAziz** [comment]
[approve ci autest]

**Comment by persiaAziz** [comment]
done @SolidWallOfCode

**Comment by zwoop** [comment]
```
./doc/developer-guide/api/functions/TSVConnAbort.en.rst
./doc/developer-guide/api/functions/TSVConnCacheObjectSizeGet.en.rst
./doc/developer-guide/api/functions/TSVConnClose.en.rst
./doc/developer-guide/api/functions/TSVConnClosedGet.en.rst
./doc/developer-guide/api/functions/TSVConnFdCreate.en.rst
./doc/developer-guide/api/functions/TSVConnIsSsl.en.rst
./doc/developer-guide/api/functions/TSVConnRead.en.rst
./doc/developer-guide/api/functions/TSVConnReadVIOGet.en.rst
./doc/developer-guide/api/functions/TSVConnReenable.en.rst
./doc/developer-guide/api/functions/TSVConnShutdown.en.rst
./doc/developer-guide/api/functions/TSVConnSslConnectionGet.en.rst
./doc/developer-guide/api/functions/TSVConnTunnel.en.rst
./doc/developer-guide/api/functions/TSVConnWrite.en.rst
./doc/developer-guide/api/functions/TSVConnWriteVIOGet.en.rst
./doc/developer-guide/api/types/TSVConnCloseFlags.en.rst
```

**Pull request merged**